File: app/src/main/java/com/weberbox/pifire/utils/TimeUtils.java
package com.weberbox.pifire.utils;

import android.annotation.SuppressLint;
import android.text.format.DateFormat;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Date;

public class TimeUtils {

    public static String getFormattedDate(String dateInMilliseconds, String dateFormat) {
        return DateFormat.format(dateFormat, Long.parseLong(dateInMilliseconds)).toString();
    }

    @SuppressLint("SimpleDateFormat")
    public static String parseDate(String date, String inputFormat, String outputFormat)
            throws ParseException {
        SimpleDateFormat sdf = new SimpleDateFormat(inputFormat);
        Date newDate = sdf.parse(date);
        sdf = new SimpleDateFormat(outputFormat);
        if (newDate != null) {
            return sdf.format(newDate);
        } else {
            return "";
        }
    }

    @SuppressLint("DefaultLocale")
    public static String formatSeconds(Integer totalSeconds) {
        if (totalSeconds != null) {
            int hours = totalSeconds / 3600;
            int minutes = (totalSeconds % 3600) / 60;
            int seconds = totalSeconds % 60;

            if (hours > 0) {
                return String.format("%02d:%02d:%02d", hours, minutes, seconds);
            } else {
                return String.format("%02d:%02d", minutes, seconds);
            }
        }
        return null;
    }

    @SuppressLint("DefaultLocale")
    public static String formatMinutes(Integer totalMinutes) {
        if (totalMinutes != null && totalMinutes > 0) {
            int hours = totalMinutes / 60;
            int minutes = totalMinutes % 60;

            return String.format("%d:%02d", hours, minutes);
        }
        return "--:--";
    }
}

File: app/src/main/java/com/weberbox/pifire/common/presentation/util/TimeUtils.kt
package com.weberbox.pifire.common.presentation.util

import android.text.format.DateFormat
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime
import java.time.format.DateTimeFormatter
import java.time.format.FormatStyle
import java.util.Locale

fun getFormattedDate(dateInMilliseconds: String, dateFormat: String): String {
    return DateFormat.format(dateFormat, dateInMilliseconds.toLong()).toString()
}

fun formatDateTimeString(
    input: String,
    inputFormat: String,
    outputFormat: String? = null,
): String {

    val inputFormatter = DateTimeFormatter.ofPattern(inputFormat)
    val outputFormatter =
        if (outputFormat == null) {
            DateTimeFormatter.ofLocalizedDateTime(FormatStyle.SHORT)
                .withLocale(Locale.getDefault())
        } else {
            DateTimeFormatter.ofPattern(outputFormat, Locale.getDefault())
        }

    return try {
        val dateTime = LocalDateTime.parse(input, inputFormatter)
        dateTime.format(outputFormatter).replace("/", "-")
    } catch (_: Exception) {
        input
    }
}

fun formatDateString(
    input: String,
    inputFormat: String,
    outputFormat: String? = null
): String {

    val inputFormatter = DateTimeFormatter.ofPattern(inputFormat)
    val outputFormatter =
        if (outputFormat == null) {
            DateTimeFormatter.ofLocalizedDate(FormatStyle.SHORT)
                .withLocale(Locale.getDefault())
        } else {
            DateTimeFormatter.ofPattern(outputFormat, Locale.getDefault())
        }

    return try {
        val date = LocalDate.parse(input, inputFormatter)
        date.format(outputFormatter).replace("/", "-")
    } catch (_: Exception) {
        input
    }
}

fun formatTimeString(
    input: String,
    inputFormat: String,
    outputFormat: String? = null
): String {

    val inputFormatter = DateTimeFormatter.ofPattern(inputFormat)
    val outputFormatter =
        if (outputFormat == null) {
            DateTimeFormatter.ofLocalizedTime(FormatStyle.SHORT)
                .withLocale(Locale.getDefault())
        } else {
            DateTimeFormatter.ofPattern(outputFormat, Locale.getDefault())
        }

    return try {
        val time = LocalTime.parse(input, inputFormatter)
        time.format(outputFormatter).replace("/", "-")
    } catch (_: Exception) {
        input
    }
}

@Suppress("unused")
fun formatSeconds(totalSeconds: Int?): String? {
    if (totalSeconds != null) {
        val hours = totalSeconds / 3600
        val minutes = (totalSeconds % 3600) / 60
        val seconds = totalSeconds % 60

        return if (hours > 0) {
            String.format(Locale.US, "%02d:%02d:%02d", hours, minutes, seconds)
        } else {
            String.format(Locale.US, "%02d:%02d", minutes, seconds)
        }
    }
    return null
}

fun formatMinutes(totalMinutes: Int?): String {
    if (totalMinutes != null && totalMinutes > 0) {
        val hours = totalMinutes / 60
        val minutes = totalMinutes % 60

        return String.format(Locale.US, "%d:%02d", hours, minutes)
    }
    return "--:--"
}
